src/js/modules/users/selectors.js

```diff
@@ -0,0 +1,45 @@
+import { createSelector } from "reselect";
+
+export const getUsers = state => state.users.users;
+export const getRoles = state => state.users.roles;
+export const getUserRoles = state => state.users.userRoles;
+
+const getRequestedContributorSlug = (state, props) => props.match.params.contributor_slug;
+const getRoleFromProps = (state, props) => props.role;
+const getUsersResponse = state => state.articles.response;
+
+export const getContributorFromSlug = createSelector(
+  [ getUsers, getRequestedContributorSlug ],
+  (users, requestedContributorSlug) => {
+    return users[ requestedContributorSlug ];
+  }
+);
+
+/**
+ * The selector returns all users within a role.
+ */
+export const getUsersWithinRole = createSelector(
```

Comment:
That function name incorrectly describes the function. But I'll rename it to getUsersInRole